tests/tlsconfig_test.go

```diff
@@ -0,0 +1,141 @@
+package tests
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	api "github.com/kubevirt/vm-console-proxy/api/v1alpha1"
+	ocpconfigv1 "github.com/openshift/api/config/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/kubevirt/vm-console-proxy/pkg/console"
+)
+
+var _ = Describe("TLS config", func() {
+	const (
+		configMapName = "vm-console-proxy"
+	)
+
+	It("should read config from ConfigMap", func() {
+		Eventually(func(g Gomega) {
+			connState, err := getTlsConnectionState()
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(connState.CipherSuite).To(BeElementOf(
+				tls.TLS_AES_128_GCM_SHA256,
+				tls.TLS_AES_256_GCM_SHA384,
+				tls.TLS_CHACHA20_POLY1305_SHA256,
+				tls.TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,
+				tls.TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,
+				tls.TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,
+				tls.TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,
+				tls.TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256,
+				tls.TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256,
+			))
+			Expect(connState.Version).To(BeNumerically(">=", tls.VersionTLS12))
+		}, 1*time.Minute, time.Second).Should(Succeed())
+	})
+
+	Context("with changed ConfigMap", func() {
+		BeforeEach(func() {
```

Comment:
What about creating the ConfigMap from scratch in `BeforeEach` and hardcoding it's values? Otherwise this might pick up some leftovers of tests if used incorrectly (in the future).

Reply:
I don't understand. How can this pickup leftovers from incorrect tests? Can you give me an example?

Reply:
I was thinking about similar restore functions in SSP tests. In the SSP tests it's not always obvious when to use the restore funtions. By creating test resources before the actual test you can be sure what to expect. Right now it is not an issue, but what if someone extends the tests in the future?

Reply:
Can you create an issue to track this in a follow up?

Reply:
https://github.com/kubevirt/vm-console-proxy/issues/5